filters/apiusagemonitoring/spec.go

```diff
@@ -288,7 +296,7 @@ func generateRegExpStringForPathTemplate(pathTemplate string) (normalizedPathTem
 		} else {
 			// this part is a placeholder: match a wildcard for it
 			matcherPathParts = append(matcherPathParts, regexUrlPathPart)
-			normalizedPathTemplateParts = append(normalizedPathTemplateParts, ":"+placeholderName)
+			normalizedPathTemplateParts = append(normalizedPathTemplateParts, fmt.Sprintf("{%s}", placeholderName))
```

Comment:
The changed pattern is incompatible to the Skipper native path pattern. Changing it this way prevents us from using the highly optimized Skipper path matcher instead of our simplified but very inefficient regex-loop implementation. This imposes the risk, that we cannot fix performance issues quickly. However, we currently have not analyzed the performance impact of our filter and thus can only assume, that the path matching will have a high worst case impact.

Reply:
I'll create an extra field in the `PathInfo` to store the converted path (`{}` instead of `:`).